client/src/components/ResourcePreview.js

```diff
@@ -34,9 +36,25 @@ function ResourcePreview(props) {
     city,
     name,
     subcategory,
+    authed,
+    updateSaved,
   } = props;
   const [src, setSrc] = useState('');
 
+  let { isSaved } = props;
+
+  const saveResourceHandler = async (_e, _id) => {
+    await saveResource(_id);
+    isSaved = true;
```

Comment:
Props shouldn't ever be set in components - you can think of them like `const` variables.  This line can be taken out since when you call `updateSaved`, it'll automatically update the `isSaved` prop for you :)